### PR TITLE
fix: [PL-22447]: set the securityPrincipal

### PR DESCRIPTION
**Pull request by bhavya181**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30578)

### CODE REVIEW

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30578)* status: 1 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/sdk/HarnessToGitPushInfoGrpcService.java, line 90 at r1](https://reviewable.io/reviews/harness/harness-core/30578#-MvNHYnlAoC1Dpd6g30v:-MvNHYnlAoC1Dpd6g30w:b-vxmloc) ([raw file](https://github.com/harness/harness-core/blob/ebe75ea1c944c20c791b462d777431c8a98fcb4a/136-git-sync-manager/src/main/java/io/harness/gitsync/sdk/HarnessToGitPushInfoGrpcService.java#L90)):*
> ```Java
>   private void setPrincipal(FileInfo request) {
>     final io.harness.security.dto.Principal principal = getPrincipal(request);
>     GlobalContextManager.upsertGlobalContextRecord(PrincipalContextData.builder().principal(principal).build());
> ```

One meaningful unit test can be we call pushInfoFuntion with the required input and then we validate that the secrurity context is set properly in thread. 

thought we might have to mock somethings but that is fine

File: 136-git-sync-manager/src/test/java/io/harness/gitsync/sdk/HarnessToGitPushInfoGrpcServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    UserPrincipal userPrincipal = UserPrincipal.newBuilder()
+                                      .setUserId(StringValue.of(name))
+                                      .setUserName(StringValue.of(userName))
+                                      .setEmail(StringValue.of(email))
+                                      .build();
+    Principal principal = Principal.newBuilder().setUserPrincipal(userPrincipal).build();
+    FileInfo fileInfo = buildFileInfo(principal);
+    GlobalContextManager.GlobalContextGuard guard = GlobalContextManager.ensureGlobalContextGuard();
+    harnessToGitPushInfoGrpcService.setPrincipal(fileInfo);
+    assertThat(SecurityContextBuilder.getPrincipal().getType()).isEqualTo(PrincipalType.USER);
```
Comment (abhinav-harness) [inline 136-git-sync-manager/src/test/java/io/harness/gitsync/sdk/HarnessToGitPushInfoGrpcServiceTest.java:54]: unset context from test too

Comment (bhavya181) [reply 136-git-sync-manager/src/test/java/io/harness/gitsync/sdk/HarnessToGitPushInfoGrpcServiceTest.java]: done

**Review by abhinav-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30578#-:-MvOZzvt6QuTwoznQjX0:bnfp4nl)* status: 2 file(s) to review. 1. unresolved discussions.

**Review by mohitg0795 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30578#-:-MvO_Au2CDon7KVvFOhN:bnfp4nl)* status: 1. unresolved discussions.

**Comment by bhavya181** [comment]
trigger messagecheck

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30578)* status: :shipit: complete! ✓

**Pull request merged**